benchmarks/cugraph-pyg/cugraph_pyg_graph_sage.py

```diff
@@ -0,0 +1,558 @@
+    from models_native import GraphSAGE
+    from torch_geometric.data import HeteroData
+    from torch_geometric.loader import NeighborLoader
+
+    import cudf
+
+    with open(os.path.join(bulk_samples_dir, 'output_meta.json'), 'r') as f:
+        output_meta = json.load(f)
+
+    dataset_path = os.path.join(output_meta['dataset_dir'], output_meta['dataset'])
+    with open(os.path.join(dataset_path, 'meta.json'), 'r') as f:
+        input_meta = json.load(f)
+
+    replication_factor = output_meta['replication_factor']
+    
+    num_edges_dict = {tuple(edge_type.split('__')): t * replication_factor for edge_type, t in input_meta['num_edges'].items()}
+    num_nodes_dict = {node_type: t * replication_factor for node_type, t in input_meta['num_nodes'].items()}
+
+    hetero_data = HeteroData()
+    num_input_features = 0
+    num_output_features = 0
+    for node_type in os.listdir(os.path.join(dataset_path, 'npy')):
+        feature_data = load_disk_features(output_meta, node_type, replication_factor=replication_factor)
+        hetero_data[node_type].x = torch.as_tensor(feature_data, device=features_device)
+
+        if feature_data.shape[1] > num_input_features:
+            num_input_features = feature_data.shape[1]
+
+        label_path = os.path.join(dataset_path, 'parquet', node_type, 'node_label.parquet')
+        if os.path.exists(label_path):
+            node_label = cudf.read_parquet(label_path)
+            if replication_factor > 1:
+                base_num_nodes = input_meta['num_nodes'][node_type]
+                dfr = cudf.DataFrame({
+                    'node': cudf.concat([node_label.node + (r * base_num_nodes) for r in range(1, replication_factor)]),
+                    'label': cudf.concat([node_label.label for r in range(1, replication_factor)]),
+                })
+                node_label = cudf.concat([node_label, dfr]).reset_index(drop=True)
+
+            node_label_tensor = torch.full((num_nodes_dict[node_type],), -1, dtype=torch.float32, device='cuda')
+            node_label_tensor[torch.as_tensor(node_label.node.values, device='cuda')] = \
+                torch.as_tensor(node_label.label.values, device='cuda')
+            
+            del node_label
+            gc.collect()
+
+            hetero_data[node_type]['train'] = (node_label_tensor > -1).contiguous()
+            hetero_data[node_type]['y'] = node_label_tensor.contiguous()
+            hetero_data[node_type]['num_nodes'] = num_nodes_dict[node_type]
+
+            num_classes = int(node_label_tensor.max()) + 1
+            if num_classes > num_output_features:
+                num_output_features = num_classes
+
+    print('done loading feature data')
+
+    # Have to load graph data for native PyG
+    parquet_path = os.path.join(
+        output_meta['dataset_dir'],
+        output_meta['dataset'],
+        'parquet'
+    )
+
+    for edge_type in os.listdir(parquet_path):
+        if re.match(r'[a-z]+__[a-z]+__[a-z]+', edge_type):
```

Comment:
I suggest using `meta.json` to iterate through edges as that is more concrete

Reply:
I'll update that here, too